docs/build/tutorials/nodes-and-staking/set-up-an-avalanche-node-with-google-cloud-platform.md

```diff
@@ -0,0 +1,139 @@
+
+### Project
+
+We will create and utilize a single GCP `Project` for deployment of all resources.
+
+#### Service Enablement
+
+Within our GCP project we will need to enable the following Cloud Services:
+- `Compute Engine`
+- `IAP`
+
+### Networking
+
+#### Compute Network
+
+We will deploy a single `Compute Network` object. This unit is where we will deploy all subsequent networking objects. It provides a logical boundary and securitization context should you wish to deploy other chain stacks or other infrastructure in GCP.
+
+#### Public IP
+
+Avalanche requires that a validator communicate outbound on the same public IP address that it advertizes for other peers to connect to it on. Within GCP this precludes the possibility of us using a Cloud NAT Router for the outbound communications and requires us to bind the public IP that we provision to the interface of the machine. We will provision a single `EXTERNAL` static IPv4 `Compute Address`.
+
+#### Subnets
+
+For the purposes of this documentation we will deploy a single `Compute Subnetwork` in the US-EAST1 `Region` with a /24 address range giving us 254 IP addresses (not all usable but for the sake of generalized documentation).
+
+### Compute
+
+#### Disk
+
+We will provision a single 400GB `PD-SSD` disk that will be attached to our VM.
+
+#### Instance
+
+We will deploy a single `Compute Instance` of size `e2-standard-8`. Observations of operations using this machine specification suggest it is memory over provisioned and could be brought down to 16GB using custom machine specification; but please review and adjust as needed (the beauty of compute virtualization!!).
+
+#### Zone
+
+We will deploy our instance into the `US-EAST1-B` `Zone`
+
+#### Firewall
+
+We will provision the following `Compute Firewall` rules:
+
+- IAP INGRESS for SSH (TCP 22) - this only allows GCP IAP sources inbound on SSH.
+- P2P INGRESS for AVAX Peers (TCP 9651)
+
+These are obviously just default ports and can be tailored to your needs as you desire.
+
+## <a name="setup-instructions"></a> Setup Instructions
+
+### GCP Account
+
+1. If you don't already have a GCP account go create one [here](https://console.cloud.google.com/freetrial)
+
+You will get some free bucks to run a trial, the trial is feature complete but your usage will start to deplete your free bucks so turn off anything you don't need and/or add a credit card to your account if you intend to run things long term to avoid service shutdowns.
+
+### Project
+
+Login to the GCP `Cloud Console` and create a new `Project` in your organization. Let's use the name `my-avax-nodes` for the sake of this setup.
+
+1. ![Select Project Dropdown](https://github.com/null-ref-ex/avax-on-gcp/blob/main/images/cloud_console_project_dropdown.png)
```

Comment:
Could you please add the images to the repository as well? By convention we add them to the `/static/img` directory, and then you can reference them as /img/image-file-name.png from the markdown code.

Reply:
Cool, makes sense, on it.